Automattic/WP-Job-Manager
# Update historical notices to use the new `render_notice` method.

Fixes #2378

### Changes proposed in this Pull Request

This PR updates historical licenses to use the new `WP_Job_Manager_Admin_Notices::render_notice` method.

This change changes the `render_notice` notice method from a `private static function` to a `public static function` and then updates the notice display per the renders at https://github.com/Automattic/WP-Job-Manager/issues/2382.

One item I have chosen to forgo for this change is the grouped update notification. I've decided this as the immediate request is to update historical notices to use the `render_notice` method, whereas the grouped update notifications would require the creation of new, non-historical notices.

One additional item I noticed is that the update notices don't inherit styling on non-WPJM pages:

![Screenshot from 2023-05-04 11-26-57](https://user-images.githubusercontent.com/2500940/236178802-553e46dc-1e27-43c2-a85d-6c1e1b5dd98d.png)

To address this I could either move the relevant `wp_enqueue_style` and `wp_register_style` outside of the `if ( in_array( $screen->id, apply_filters(....` conditional, or, add more screen IDs to the check. I've opted to move the style as it's not possible to anticipate all of the potential screens the notices might display on, or even if it was, the check would be so long as to be pointless.

To prevent a conflict with WC styling, I've also made use of a `! important` declaration. I don't know if, instead, we want to copy into WPJM the relevant `notice` styles and have our own `wpjm-notice` class instead. I'm not keen on `! important` so if we want to go to in that direction, I'm all for it. The declaration does seem the most straightforward though, and it's specific to our notices.

Without this change, notices lose their padding on certain WC pages (like WooCommerce --> Orders). For example:

![Screenshot from 2023-05-04 11-59-24](https://user-images.githubusercontent.com/2500940/236185464-7e9cd454-27c2-4ebb-afad-c17964ea65c0.png)

### Testing instructions

* Checkout branch
* Install a couple of paid WPJM plugins, but don't input a license key
* Notices should display with the updated formatting as shown below
* If you've tested this prior and an update notice isn't showing, use `wp option delete wp_job_manager_dismissed_notices` through WP CLI (or otherwise delete this option) to force the notice to display.

![Screenshot from 2023-05-04 11-03-16](https://user-images.githubusercontent.com/2500940/236173881-e769c390-cd8d-4580-9fb5-3d418764e9e0.png)

* To show an error notice, comment out the code to show the missing license notice https://github.com/Automattic/WP-Job-Manager/blob/15393a2e1a84d2d194f0097d2023e2940ac38e4e/includes/helper/class-wp-job-manager-helper.php#L569-L576
* change the `elseif` into an `if`, and then the check to show an error notice from `! empty` to `empty` https://github.com/Automattic/WP-Job-Manager/blob/15393a2e1a84d2d194f0097d2023e2940ac38e4e/includes/helper/class-wp-job-manager-helper.php#L576-L583
